LPS-197957  LPS-199722

lfbesada opened the pull request.

sandrodw3 commented:
Pull request submitted to https://github.com/p2kmgcl/liferay-portal/pull/159. [See changes here.](https://github.com/sandrodw3/liferay-portal/compare/c30e1f991af2~3...c30e1f991af2) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

lfbesada committed:
LPS-196807 Add a method to get the AssetEntry from the base model

lfbesada committed:
LPS-196807 Override the new method in JournalArticleAssetRendererFactory to take into account that an article could have none, one or more than one asset entries

lfbesada committed:
LPS-196807 Override the new method in the depot wrapper

lfbesada committed:
LPS-196807 Fix generic class types

lfbesada committed:
LPS-196807 Use the new method to get the entryClassPK from the base model

Calculates correctly the entryClassPK by delegating the calculation to the asset renderer factory. Also, consider that the asset entry could not exist, for example, in the case of published versions that are not the latest ones.

The entry class PK of the resource models does not always match the resourcePrimKey. There could be more than one asset entry from the same article, which happens in the case of non-visible versions (draft, scheduled, pending...).

In this scenario, only the last approved version, or the first one in case no published version, will use the article resourcePrimKey as her asset entryClassPK. The other non-visible versions will use its ID as entryClassPK.

lfbesada committed:
LPS-196807 Adds the missing fields if not present to handle those articles that do not have asset entry

lfbesada committed:
LPS-196807 Auto

lfbesada committed:
LPS-196807 SF simplify

lfbesada committed:
LPS-196807 Add an integration test to assert that all versions, approved or not, are indexed using the right content

lfbesada committed:
LPS-196807 Add an integration test to assert that when changing the categorization on a draft article each version will have the correct category name indexed

lfbesada committed:
LPS-196807 Add an integration test to assert that if we change the article tag classification without approving the article each version will have the proper tag name indexed

lfbesada committed:
LPS-196807 Ensure these tests are always running with indexAllArticleVersions enabled and restore the original value after class

lfbesada committed:
LPS-196807 SF, rename, this test class is related to the fields added to the article document because of its AssetEntry's relations and not only when reindexing

lfbesada committed:
LPS-196807 Override the new method in other AssetRendererFactories in order to prevent getting twice the model

sandrodw3 committed:
LPS-197957 Rename to getUsersURL

sandrodw3 committed:
LPS-197957 Move actions and conditions states to RulesModal component

sandrodw3 committed:
LPS-197957 Add action command to add a rule

sandrodw3 committed:
LPS-197957 Add action command to delete a rule

sandrodw3 committed:
LPS-197957 Add action command to update a rule

sandrodw3 committed:
LPS-197957 Send URLs

sandrodw3 committed:
LPS-197957 Add thunks to add, update and delete a rule

sandrodw3 committed:
LPS-197957 Dispatch thunks instead of actions

victorg1991 committed:
LPS-197957 Convert draft service fetch to typescript

victorg1991 committed:
LPS-197957 Types

sandrodw3 committed:
Aclaraciones para Lourdes

lfbesada committed:
LPS-199722 Adds Page Rule Definition to rest-openapi.yaml

lfbesada committed:
LPS-199722 Adds LayoutStructureRule to handle the page rules

lfbesada committed:
LPS-199722 BuildRest

lfbesada committed:
LPS-199722 Adds the page rules to the layout structure

lfbesada committed:
LPS-199722 Implements page rules export

lfbesada committed:
LPS-199722 Implements page rules import

lfbesada committed:
LPS-199722 Implements page rules export

lfbesada committed:
LPS-199722 Adds the page rules to page definition json schema

lfbesada committed:
LPS-199722 LPS-199722 Implements page rules export

lfbesada committed:
LPS-199722 Adds the page rules to the layout structure

lfbesada committed:
LPS-199722 Baseline

lfbesada committed:
LPS-XXX Implement add rule action

lfbesada committed:
LPS-XXX Implement delete rule action

lfbesada committed:
LPS-XXX Implement update rule action

lfbesada committed:
LPS-199722 Adds the page rules to the layout structure

lfbesada committed:
LPS-XXXX prevents NPE when exception occurs during validation

lfbesada committed:
LPS-199722 Adds the page rules to the layout structure
